server/controllers/recall/deleteRecallController.ts

```diff
@@ -0,0 +1,122 @@
+import { Request, Response } from 'express'
+// eslint-disable-next-line import/no-unresolved
+import { Recall } from 'models'
+import logger from '../../../logger'
+import { Services } from '../../services'
+import PrisonService from '../../services/PrisonService'
+import { SessionManager } from '../../services/sessionManager'
+
+async function addLocationNameToRecall(recall: Recall, prisonService: PrisonService, username: string) {
+  if (!recall) return recall
+
+  let locationName: string | null = null
+
+  if (recall.location) {
+    try {
+      const prisonNames = await prisonService.getPrisonNames([recall.location], username)
+      locationName = prisonNames.get(recall.location) || null
+    } catch (err) {
+      locationName = null
+      logger.error(err)
+    }
+  }
+
+  return {
+    ...recall,
+    locationName,
+  }
+}
+
+export default class DeleteRecallController {
+  constructor(private readonly services: Services) {}
+
+  /**
+   * GET /person/:nomisId/recall/:recallId/delete
+   * Renders the delete confirmation page for a recall
+   */
+  async get(req: Request, res: Response): Promise<void> {
+    const { nomisId, recallId } = req.params
+    const { from } = req.query
+    const { user } = res.locals
+    const username = user?.username
+
+    const recall = await this.services.recallService.getRecall(recallId, username)
+    const enrichedRecall = await addLocationNameToRecall(recall, this.services.prisonService, username)
+
+    const prisoner = await this.services.prisonerService.getPrisonerDetails(nomisId, username)
```

Comment:
isnt the prisoner meant to be in res.locals already?

there's a middleware function that does that by the looks of it - shouldnt you be using the loadPrisoner middleware